pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -0,0 +1,1089 @@
+		pvcName := name + fmt.Sprintf("%d", i)
+		err := Mgr.GetClient().Get(context.TODO(),
+			types.NamespacedName{Name: pvcName, Namespace: p.Namespace}, pvc)
+		if err != nil {
+			return err
+		}
+		pvc.SetOwnerReferences(*ownerRef)
+		err = Mgr.GetClient().Update(context.TODO(), pvc)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func migratePVC(p *v1alpha1.PravegaCluster, b *bkapi.BookkeeperCluster) error {
+	err := movePVC("index", p, b)
+	if err != nil {
+		return err
+	}
+	log.Printf("Updating owner reference for index PVCs")
+	err = movePVC("journal", p, b)
+	if err != nil {
+		return err
+	}
+	log.Printf("Updating owner reference for journal PVCs")
+	err = movePVC("ledger", p, b)
+	if err != nil {
+		return err
+	}
+	log.Printf("Updating owner reference for ledger PVCs")
+	return nil
+}
+
+func migratePDB(p *v1alpha1.PravegaCluster, b *bkapi.BookkeeperCluster) error {
+	ownerRef := &[]metav1.OwnerReference{
+		{
+			APIVersion:         "bookkeeper.pravega.io/v1alpha1",
+			Kind:               "BookkeeperCluster",
+			Name:               b.GetName(),
+			UID:                b.GetUID(),
+			BlockOwnerDeletion: pointer.BoolPtr(true),
+			Controller:         pointer.BoolPtr(true),
+		},
+	}
+	pdb := &policyv1beta1.PodDisruptionBudget{}
+	name := nameForBookie(p.Name)
+	log.Printf("Updating owner reference for PDB %s", name)
+	err := Mgr.GetClient().Get(context.TODO(), types.NamespacedName{Name: name, Namespace: p.Namespace}, pdb)
+	if err != nil {
+		return err
+	}
+	pdb.SetOwnerReferences(*ownerRef)
+	return Mgr.GetClient().Update(context.TODO(), pdb)
+}
+
+func specCopy(srcObj *v1alpha1.PravegaCluster, b *bkapi.BookkeeperCluster) {
+	bkSpec := srcObj.Spec.Bookkeeper
+	if bkSpec.Image != nil {
+		b.Spec.Image.PullPolicy = bkSpec.Image.PullPolicy
+		b.Spec.Image.Repository = bkSpec.Image.Repository
+		b.Spec.Image.Tag = bkSpec.Image.Tag
+	}
+
+	b.Spec.Replicas = bkSpec.Replicas
+
+	if bkSpec.Storage != nil {
+		b.Spec.Storage.IndexVolumeClaimTemplate = bkSpec.Storage.IndexVolumeClaimTemplate
+		b.Spec.Storage.JournalVolumeClaimTemplate = bkSpec.Storage.JournalVolumeClaimTemplate
+		b.Spec.Storage.LedgerVolumeClaimTemplate = bkSpec.Storage.LedgerVolumeClaimTemplate
+	}
+
+	if bkSpec.AutoRecovery != nil {
+		b.Spec.AutoRecovery = bkSpec.AutoRecovery
+	}
+
+	b.Spec.ServiceAccountName = bkSpec.ServiceAccountName
+
+	if bkSpec.Resources != nil {
+		b.Spec.Resources = bkSpec.Resources
+	}
+
+	if bkSpec.Options != nil {
+		b.Spec.Options = bkSpec.Options
+	}
+
+	if bkSpec.BookkeeperJVMOptions != nil {
+		b.Spec.JVMOptions.ExtraOpts = bkSpec.BookkeeperJVMOptions.ExtraOpts
+		b.Spec.JVMOptions.GcLoggingOpts = bkSpec.BookkeeperJVMOptions.GcLoggingOpts
+		b.Spec.JVMOptions.GcOpts = bkSpec.BookkeeperJVMOptions.GcOpts
+		b.Spec.JVMOptions.MemoryOpts = bkSpec.BookkeeperJVMOptions.MemoryOpts
+	}
+
+	b.Spec.ZookeeperUri = srcObj.Spec.ZookeeperUri
+	// name of config-map having pravega configuration
+	b.Spec.EnvVars = "pravega-config"
```

Comment:
why are we hardcoding envar to `pravega-config`

Reply:
The name of the config map does not matter. 
Please refer: https://github.com/pravega/bookkeeper-operator/blob/master/doc/manual-installation.md#install-the-bookkeeper-cluster-manually